server/app/lib/actions/fusor/deployment/openstack/cfme_upload.rb

```diff
@@ -0,0 +1,85 @@
+#
+# Copyright 2015 Red Hat, Inc.
+#
+# This software is licensed to you under the GNU General Public
+# License as published by the Free Software Foundation; either version
+# 2 of the License (GPLv2) or (at your option) any later version.
+# There is NO WARRANTY for this software, express or implied,
+# including the implied warranties of MERCHANTABILITY,
+# NON-INFRINGEMENT, or FITNESS FOR A PARTICULAR PURPOSE. You should
+# have received a copy of GPLv2 along with this software; if not, see
+# http://www.gnu.org/licenses/old-licenses/gpl-2.0.txt.
+
+require 'fog'
+
+module Actions
+  module Fusor
+    module Deployment
+      module OpenStack
+        #Upload CFME to the overcloud
+        class CfmeUpload < Actions::Base
+          def humanized_name
+            _('Upload CFME to the overcloud')
+          end
+
+          def plan(deployment)
+            plan_self(deployment_id: deployment.id)
+          end
+
+          def run
+            Rails.logger.debug '====== CFME Upload run method ======'
+            deployment = ::Fusor::Deployment.find(input[:deployment_id])
+            overcloud = { :openstack_auth_url  => "http://#{deployment.openstack_overcloud_address}:5000/v2.0/tokens",
+                          :openstack_username  => 'admin', :openstack_tenant => 'admin',
+                          :openstack_api_key   => deployment.openstack_overcloud_password }
+            cfme_image = download_cfme_image
+            upload_cfme_image(overcloud, "#{deployment.name}-cfme", cfme_image)
+            Rails.logger.debug '====== Leaving CFME Upload run method ======'
+          end
+
+          def cfme_upload_completed
+            Rails.logger.info 'CFME Upload Completed'
+          end
+
+          def cfme_upload_failed
+            fail _('CFME Upload failed')
+          end
+
+          private
+
+          def download_cfme_image
+            sat_url = Setting[:foreman_url].gsub('https://', 'http://')
+            ####FIX CFME IMAGE
+            image_url = "#{sat_url}/pulp/isos/Default_Organization-Red_Hat_CloudForms-Red_Hat_CloudForms"\
```

Comment:
We'll need to add in the work brad had of looking up the latest CFME image.

Perhaps we do the look up in a higher level class and just pass in the image path/filename at this phase.